internal/rpc/jsonrpc/methods.go

```diff
@@ -1413,6 +1416,42 @@ func (s *Server) getMasterPubkey(ctx context.Context, icmd interface{}) (interfa
 	return masterPubKey.String(), nil
 }
 
+// getPeerInfo responds to the getpeerinfo request.
+// It gets the network backend and views the data on remote peers when in spv mode
+func (s *Server) getPeerInfo(ctx context.Context, icmd interface{}) (interface{}, error) {
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, errUnloadedWallet
+	}
+	n, err := w.NetworkBackend()
+
+	if err != nil {
+		return nil, err
+	}
+
+	syncer, ok := n.(*spv.Syncer)
+
+	if ok {
+		infos := make([]*types.GetPeerInfoResult, 0, len(syncer.GetRemotePeers()))
+		for _, rp := range syncer.GetRemotePeers() {
+			snapshot := rp.StatsSnapshot()
+			info := &types.GetPeerInfoResult{
+				ID:             int32(snapshot.Id),
+				Addr:           snapshot.Raddr,
+				Services:       fmt.Sprintf("%08d", uint64(snapshot.Services)),
+				StartingHeight: int64(snapshot.InitHeight),
+				Version:        snapshot.Version,
+				SubVer:         snapshot.UserAgent,
+				BanScore:       snapshot.Banscore,
+			}
+			infos = append(infos, info)
+		}
+		return infos, nil
+	} else {
+		return nil, nil
```

Comment:
This control flow is too nested.  Return early in the case that the syncer is not a SPV syncer instead.

Reply:
fixed